modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/CustomMustacheFactory.java

```diff
@@ -253,6 +258,42 @@ static boolean match(String variable) {
         }
     }
 
+    /**
+     * This function encodes a string using the {@link URLEncoder#encode(String, String)} method
+     * with the UTF-8 charset.
+     */
+    static class UrlEncoderCode extends DefaultMustache {
```

Comment:
As a follow up PR I'd like to externalize the function into dedicated classes along with tests, factories etc because we now have multiple functions. This will make the code more readable.